### PR TITLE
updated rp2-docs-quickref

### PR DESCRIPTION
**Pull request by NitiKaur**

### CODE REVIEW

**Comment by mattytrentini** [comment]
Looks good @NitiKaur but please add another commit that wraps the column length to ~80.

**Comment by robert-hh** [comment]
Actually there are two UARTs, UART0 and UIART1. UART0 can be mapped to GPIO 1/2, 12/13 and 16/17, and UART1 to GPIO 6/7 and 8/9. Since for Using the GPIO numbers have to be used. Also, the example in the following lines is wrong. It should be
```
    from machine import UART, Pin
    uart1 = UART(1, baudrate=9600, tx=Pin(4), rx=Pin(5))
    uart1.write('hello')  # write 5 bytes
    uart1.read(5)         # read up to 5 bytes
```

**Comment by robert-hh** [comment]
I also noticed that the example for ADC is wrong. It should e.g. be:
```
from machine import ADC, Pin
adc = ADC(Pin(26))          # create ADC object on ADC pin
adc.read_u16()              # read value, 0-65535 across voltage range 0.0v - 3.3v
```
Please take you time to test the examples before putting them into the document.

**Comment by robert-hh** [comment]
The examples for SPI and I2C also need a fix.
HW SPI:
```
from machine import Pin, SPI

spi = SPI(1, 10000000)  # Default assignment: sck=Pin(10), mosi=Pin(11), miso=Pin(8) 
spi = SPI(1, 10000000, sck=Pin(14), mosi=Pin(15), miso=Pin(12))
spi = SPI(0, baudrate=80000000, polarity=0, phase=0, bits=8, sck=Pin(6), mosi=Pin(7), miso=Pin(4))
```
Hardware I2C:
```
from machine import Pin, I2C
i2c = I2C(0)   # default assignment: scl=Pin(8), sda=Pin(9)
i2c = I2C(1, scl=Pin(3), sda=Pin(2), freq=400000)
```

**Comment by dpgeorge** [comment]
I think the default pins for hardware I2C(0) are scl=Pin(9) and sda=Pin(8).

**Comment by robert-hh** [comment]
Right. Changed.

**Comment by NitiKaur** [comment]
Thank You @robert-hh . I have made the corrections as suggested

File: docs/rp2/quickref.rst [hunk 1]
```diff
@@ -59,7 +60,18 @@ Use the :mod:`time <utime>` module::
 Timers
 ------
 
-How do they work?
+RP2040's system timer peripheral provides a global microsecond timebase and
+generates interrupts for the same. The software timer is available currently,
+and there are unlimited number of them available(limited by memory). There is no
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:65]:
```suggestion
and there are unlimited number of them available (limited by memory). There is no
```

File: docs/rp2/quickref.rst [hunk 2]
```diff
@@ -84,19 +96,35 @@ Use the :ref:`machine.Pin <machine.Pin>` class::
 UART (serial bus)
 -----------------
 
-See :ref:`machine.UART <machine.UART>`. ::
+There are two UARTs, UART0 and UIART1. UART0 can be mapped to GPIO 1/2, 12/13
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:99]:
```suggestion
There are two UARTs, UART0 and UART1. UART0 can be mapped to GPIO 1/2, 12/13
```

File: docs/rp2/quickref.rst [hunk 3]
```diff
@@ -112,14 +140,18 @@ Use the ``machine.PWM`` class::
 ADC (analog to digital conversion)
 ----------------------------------
 
-How does the ADC module work?
+RP2040 has five ADC channels in total, four of which are 12-bit SAR based ADCs:
+GP26, GP27, GP28 and GP29. The input signal for ADC0, ADC1 and ADC2 can be
+connected with GP26,GP27,GP28. The fourth is used to measure the VSYS voltage on
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:145]:
```suggestion
connected with GP26, GP27, GP28. The fourth is used to measure the VSYS voltage on
```

File: docs/rp2/quickref.rst [hunk 4]
```diff
@@ -112,14 +140,18 @@ Use the ``machine.PWM`` class::
...
+the board.The standard ADC range is 0-3.3V. The fifth channel is connected to
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:146]:
```suggestion
the board. The standard ADC range is 0-3.3V. The fifth channel is connected to
```

File: docs/rp2/quickref.rst [hunk 5]
```diff
@@ -84,19 +96,35 @@ Use the :ref:`machine.Pin <machine.Pin>` class::
...
+ 	
+	REPL over UART is disabled by default. You can see the :ref:`rp2_intro` for
+	details on how to enable REPL over UART.
+
 
 PWM (pulse width modulation)
 ----------------------------
 
-How does PWM work on the RPi RP2xxx?
+There are 8 independent channels each of which have two outputs making it 16 PWM
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:126]: If I'm being super-picky the old-school rule is to write the words for numbers less than ten, digits for 10 and above. The more modern take is to just write digits.

I'm not rigid on either rule but there's '8' and 'two' in this one sentence - which is a little inconsistent either way!

File: docs/rp2/quickref.rst [hunk 6]
```diff
@@ -156,14 +188,14 @@ Software SPI (using bit-banging) works on all pins, and is accessed via the
...
+the same methods as software SPI above::
 
     from machine import Pin, SPI
 
-    spi = SPI(1, 10000000)
-    spi = SPI(1, 10000000, sck=Pin(14), mosi=Pin(13), miso=Pin(12))
-    spi = SPI(2, baudrate=80000000, polarity=0, phase=0, bits=8, firstbit=0, sck=Pin(18), mosi=Pin(23), miso=Pin(19))
+    spi = SPI(1, 10000000)  # Default assignment: sck=Pin(10), mosi=Pin(11), miso=Pin(8) 
+    spi = SPI(1, 10000000, sck=Pin(14), mosi=Pin(15), miso=Pin(12))
+    spi = SPI(0, baudrate=80000000, polarity=0, phase=0, bits=8, sck=Pin(6), mosi=Pin(7),         miso=Pin(4))
```
Comment (dpgeorge) [inline docs/rp2/quickref.rst:198]: remove space before `miso`

**Comment by NitiKaur** [comment]
@dpgeorge I think this PR is up for  final merge

File: docs/rp2/quickref.rst [hunk 7]
```diff
@@ -112,14 +140,18 @@ Use the ``machine.PWM`` class::
 ADC (analog to digital conversion)
 ----------------------------------
 
-How does the ADC module work?
+RP2040 has five ADC channels in total, four of which are 12-bit SAR based ADCs:
+GP26, GP27, GP28 and GP29. The input signal for ADC0, ADC1 and ADC2 can be
+connected with GP26, GP27, GP28. The fourth is used to measure the VSYS voltage on
+the board. The standard ADC range is 0-3.3V. The fifth channel is connected to
+the in-built temperature sensor and can be used for measuring the temperature by
+reading the analog values.
```
Comment (dpgeorge) [inline docs/rp2/quickref.rst:148]: I couldn't find anything about VSYS on the 4th channel in the datasheet.  As far as I understand, channels ADC0-ADC3 can be connected with GP26-GP29 (inclusive) and the 5th channel is for temperature.

File: docs/rp2/quickref.rst [hunk 8]
```diff
@@ -156,15 +188,16 @@ Software SPI (using bit-banging) works on all pins, and is accessed via the
 Hardware SPI bus
 ----------------
 
-Hardware SPI is accessed via the :ref:`machine.SPI <machine.SPI>` class and
-has the same methods as software SPI above::
+Hardware SPI is accessed via the :ref:`machine.SPI <machine.SPI>` class and has
+the same methods as software SPI above::
```
Comment (dpgeorge) [inline docs/rp2/quickref.rst:192]: How many hardware SPI buses are available?

File: docs/rp2/quickref.rst [hunk 9]
```diff
@@ -202,13 +234,16 @@ See :ref:`machine.RTC <machine.RTC>` ::
...
     rtc.datetime() # get date and time
 
 WDT (Watchdog timer)
 --------------------
 
-Is there a watchdog timer?
+The RP2XXX port has a watchdog which is a countdown timer that can restart parts
+of the chip if it reaches zero. This helps in restarting the processor if
+software gets stuck in an infinite loop. The programmer must periodically write
+a value to the watchdog to stop it from reaching zero.
```
Comment (dpgeorge) [inline docs/rp2/quickref.rst:246]: The user doesn't need to write a value to the watchdog, they just need to call the feed() method (which will probably write a register but that's an implementation detail that doesn't need to be explained here).

**Comment by dpgeorge** [comment]
> I have corrected the typos. I think this PR is up for merge

Thank you, it's very close now, just a few minor points, see above.

**Comment by dpgeorge** [comment]
Superseded by #7519.

**Pull request closed without merging**